icu4j/main/classes/core/src/com/ibm/icu/message2/Mf2ParseException.java

```diff
@@ -0,0 +1,73 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2;
+
+/**
+ * An exception thrown when parsing a {@link MessageFormatter} syntax fails.
+ *
+ * @internal ICU 72 technology preview
+ * @deprecated This API is for ICU internal use only.
+ */
+@Deprecated
+public class Mf2ParseException extends RuntimeException {
```

Comment:
Note: The exceptions are public API but not in the design doc / proposal. We will need to amend the proposal to include all public API.

Reply:
I've checked how I use the exceptions.

I used the `Mf2ParseException` to report parsing errors (missing `{` and so on), when one creates a `MessageFormatter` from a string.
But now it is replaced with `IllegalArgumentException`, same as MF1

And I used `Mf2FormatException` (now `MessageFormatterException`) when the `mf.format` is invoked.
Thinks like type problems, or bad values for the function options.

But it looks like MF1 uses `IllegalArgumentException` for that too (for example I pass a String to a "...{foo, date}...")
Which kind of is true.

Idea: should I use `IllegalArgumentException` instead of `MessageFormatterException` and remove it?
That would make it more consistent with MF1.
And makes it easier to "split" into a backport that is not ICU4J dependent :-)

Reply:
OK, use IllegalArgumentException and remove this class.

Reply:
Done.

Throwing IllegalArgumentException, added that info to javadoc, removed this class.